### DISCUSSION TITLE
SchemaField.Array 组件，当点击添加的时候可以监听到添加事件？

### DISCUSSION THREAD

**Discussion by AganStr** [alibaba/formily#1251] [Q&A 2.x] [answered]
<SchemaField.Array
  x-decorator='FormItem'
  x-component='ArrayItems'>
  <SchemaField.Object x-component='ArrayItems.Item'>
    <SchemaField.Void x-component='Space'>
      <SchemaField.Void x-decorator='FormItem' x-component='ArrayItems.Remove' />
      <SchemaField.Void x-decorator='FormItem' x-component='ArrayItems.MoveUp' />
      <SchemaField.Void x-decorator='FormItem' x-component='ArrayItems.MoveDown' />
    </SchemaField.Void>
  </SchemaField.Object>
<SchemaField.Void x-component='ArrayItems.Addition' title='添加' />
</SchemaField.Array>

**Comment by janryWang** [comment, marked as answer]
x-component-props.onClick就是添加事件

**Reply by AganStr** [reply]
<SchemaField.Void 
x-component='ArrayItems.Addition' 
title='添加' 
x-component-props={{
    onClick: (res) => {
	    console.log(2222);
    },
}} />
这样不行啊

**Reply by janryWang** [reply]
看了下，还没支持，等下个版本吧

**Reply by AganStr** [reply]
好的

**Reply by blakeyi** [reply]
最新版本似乎已经支持了
在自定义组件内  emit('transfer',value.value)
在jsonschema里的
                    'x-component': 'FormPreviewer',
                    'x-component-props': {
                        onTransfer: (value) => {
                            console.log(value)
                        }
                    },

**Comment by AganStr** [comment]
这是来自QQ邮箱的假期自动回复邮件。
 
您好，我最近正在休假中，无法亲自回复您的邮件。我将在假期结束后，尽快给您回复。
